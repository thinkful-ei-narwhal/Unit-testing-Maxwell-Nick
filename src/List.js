import React from 'react';
import Card from './Card';
import STORE from './store';
import './listStyles.css'

export default function List(props) {
  const cardComponents = props.cards.map(cardId => {
    const card = STORE.allCards[cardId];
    return <Card key={card.id} title={card.title} content={card.content} />;
  });

  return (
    <div className='List'>
      <header>
        {props.header}
      </header>
      <div className='List-cards'>
        {cardComponents}
      </div>
    </div>
  );
}
