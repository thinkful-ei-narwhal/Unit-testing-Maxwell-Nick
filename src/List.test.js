import React from 'react';
import STORE from './store';
import List from './List';
import library from 'react-test-renderer';

it('renders the UI as expected', () => {
  const jsonStringNew = library
    .create(<List key={STORE.lists[0].id} header={STORE.lists[0].header} cards={STORE.lists[0].cardIds} />)
    .toJSON();
  expect(jsonStringNew).toMatchSnapshot();
});