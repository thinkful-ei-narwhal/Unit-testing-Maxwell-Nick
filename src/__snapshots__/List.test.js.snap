// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="List"
>
  <header>
    First list
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <h3>
        First card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <h3>
        Second card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <h3>
        Fifth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <h3>
        Sixth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <h3>
        Seventh card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <h3>
        Tenth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <h3>
        Twelfth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <h3>
        Thirteenth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
  </div>
</div>
`;
